superset-frontend/plugins/plugin-chart-cartodiagram/src/components/ChartLayer.tsx

```diff
@@ -76,9 +71,17 @@ export class ChartLayer extends Layer {
       this.chartBackgroundBorderRadius = options.chartBackgroundBorderRadius;
     }
 
-    if (options.prerenderedZoomLevels) {
-      this.prerenderedZoomLevels = options.prerenderedZoomLevels;
-    }
+    this.loadingMask = document.createElement('div');
+    this.loadingMask.style.position = 'relative';
+    this.loadingMask.style.height = '100%';
+    const spinner = document.createElement('img');
+    spinner.src = Loader;
+    spinner.style.position = 'relative';
+    spinner.style.width = '50px';
+    spinner.style.top = '50%';
+    spinner.style.left = '50%';
+    spinner.style.transform = 'translate(-50%, -50%)';
+    this.loadingMask.appendChild(spinner);
```

Comment:
I find it easier to read like this

```suggestion
    const spinner = document.createElement('img');
    spinner.src = Loader;
    spinner.style.position = 'relative';
    spinner.style.width = '50px';
    spinner.style.top = '50%';
    spinner.style.left = '50%';
    spinner.style.transform = 'translate(-50%, -50%)';
    
    this.loadingMask = document.createElement('div');
    this.loadingMask.style.position = 'relative';
    this.loadingMask.style.height = '100%';
    this.loadingMask.appendChild(spinner);
```